compiler/src/dotty/tools/dotc/core/quoted/PickledQuotes.scala

```diff
@@ -143,7 +143,7 @@ object PickledQuotes {
             case tree: Ident if tree.symbol == paramSym => x1Ref().withPos(tree.pos)
             case _ => super.transform(tree)
           }
-        }.transform(ddef.rhs)
+        }.transform(ddef.rhs).changeOwner(ddef.symbol, ctx.owner)
```

Comment:
To be more efficient, you can replace your TreeMap by a TreeTypeMap(oldOwners = ..., newOwners = ...), and avoid the need for changeOwner (which would create its own TreeTypeMap)

Reply:
Will change it